### ISSUE TITLE
Guzzle error when trying to use this plugin

### ISSUE THREAD

**Issue by ergonlogic** [php-tuf/composer-integration#33]
In the context of the Drupal Association TUF project, I'm setting up end-to-end tests for the TUF metadata that we're generating with Rugged. I'm using Satis to generate a Composer package repository that works nicely to proxy `https://packages.drupal.org/8`.

Upon enabling TUF verification (note that TUF metadata is not yet present), I get the following error:

```
PHP Fatal error:  Uncaught Error: Call to undefined function GuzzleHttp\choose_handler() in /var/www/html/d9-site/vendor/guzzlehttp/guzzle/src/HandlerStack.php:42
Stack trace:
#0 /var/www/html/d9-site/vendor/guzzlehttp/guzzle/src/Client.php(65): GuzzleHttp\HandlerStack::create()
#1 /var/www/html/d9-site/vendor/php-tuf/php-tuf/src/Client/GuzzleFileFetcher.php(72): GuzzleHttp\Client->__construct(Array)
#2 /var/www/html/d9-site/vendor/php-tuf/composer-integration/src/TufValidatedComposerRepository.php(47): Tuf\Client\GuzzleFileFetcher::createFromUri('http://packages...')
#3 /var/www/html/d9-site/vendor/php-tuf/composer-integration/src/Plugin.php(168): Tuf\ComposerIntegration\TufValidatedComposerRepository->__construct(Array, Object(Composer\IO\ConsoleIO), Object(Composer\Config), Object(Composer\Util\HttpDownloader), Object(Composer\EventDispatcher\EventDispatcher))
#4 /var/www/html/d9-site/vendor/php-tuf/composer-integration/src/Plugin.php(123): Tuf\ComposerIntegration\Plugin->addTufValidationToRepositories(Object(Composer\Compo in /var/www/html/d9-site/vendor/guzzlehttp/guzzle/src/HandlerStack.php on line 42
```

Here's the [composer.json](https://github.com/php-tuf/composer-integration/files/8210006/composer.json.txt).

Note that without the `"tuf": true`, `composer install` works just fine, emitting:
> Authenticity of packages from http://packages.ddev.site are not verified by TUF.

**Comment by ergonlogic** [comment]
It looks like the includes of some function files aren't working. For now, I've just included these directly (which is kinda gross, but :shrug: ). See ergonlogic/composer-integration@7eb2e53

**Comment by ergonlogic** [comment]
I started seeing this with a number of other dependencies. It looked like the system Composer isn't aware of the plugin-specific dependencies. It doesn't load the project autoloader. Once I installed and used a project-specific Composer, things started working properly.

Digging a little deeper, I found that, indeed, [Composer pluging autoloading](https://getcomposer.org/doc/articles/plugins.md#plugin-autoloading) works a little differently:
> Due to plugins being loaded by Composer at runtime, and to ensure that plugins which depend on other packages can function correctly, a runtime autoloader is created whenever a plugin is loaded. That autoloader is only configured to load with the plugin dependencies, so you may not have access to all the packages which are installed.

The problems here all have to do with missing function declarations. This is handled using the [files autoloading mechanism](https://getcomposer.org/doc/04-schema.md#files). But there's a bug: [composer#10024: Composer does not respect plugin files autoload requirements](https://github.com/composer/composer/issues/10024)

It looks like a fix was merged a few months ago, and made it into the latest release [2.3.0-RC2](https://github.com/composer/composer/releases/tag/2.3.0-RC2)
